src/Confluent.Kafka/Internal/CancellablePollThread.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright 2016-2017 Confluent Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Refer to LICENSE for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+
+
+namespace Confluent.Kafka
+{
+    internal class CancellablePollThread
+    {
+        private readonly int millisecondsTimeout;
+        private readonly Object startStopLock = new Object();
+
+        private CancellationTokenSource consumerCts = null;
```

Comment:
As the fields are initialized to `null` by default, isn't the assignment redundant?